core/network/besu/contracts/interop/manageAssetAny.sol

```diff
@@ -57,18 +58,65 @@ contract AssetExchangeContract is ERC1155Holder {
         bytes32 indexed lockContractId
     );
 
-    // The sender locks a  asset with a hash lock and an expiry time
+    function bytesToAddress(bytes memory bys)
+        public
+        pure
+        returns (address addr)
+    {
+        assembly {
+            addr := mload(add(bys, 20))
+        }
+    }
+
+    function bytesTobytes32(bytes memory bys)
+        public
+        pure
+        returns (bytes32 res)
+    {
+        assembly {
+            res := mload(add(bys, 32))
+        }
+    }
+
+    function stringToUint(string memory s) internal returns (uint256 result) {
+        bytes memory b = bytes(s);
+        uint256 i;
+        result = 0;
+        for (i = 0; i < b.length; i++) {
+            uint256 c = uint256(uint8(b[i]));
+            if (c >= 48 && c <= 57) {
+                result = result * 10 + (c - 48);
+            }
+        }
+    }
+
     function lockAsset(
-        address receiver,
+        bytes memory rawParams,
```

Comment:
may be call this `assetExchangeAgreement`? Also do you want to remove amount parameter as well? Because if its fungible lock, this parameter would contain `FungibleAssetExchangeAgreement`.

For Hybrid cases, I thought we discussed to declare new protobuf message called `HybridAssetExchangeAgreement` where both AssetID and quantity fields would be present.

May be have a flag as input to indicate which case this is (i.e. whether fungible, non-fungible, or hybrid, to decide to which protobuf to decode)